eyelid-match-ops/src/primitives/poly/modular_poly/modulus.rs

```diff
@@ -86,7 +84,29 @@ pub fn mod_poly_ark_ref(dividend: &Poly) -> Poly {
 ///
 /// This uses an [`ark-poly`] library implementation, and entirely replaces the inner polynomial representation.
 #[cfg(inefficient)]
-pub fn mod_poly_ark_mut(dividend: &mut Poly) {
+pub fn mod_poly_ark_mut<const MAX_POLY_DEGREE: usize>(dividend: &mut Poly<MAX_POLY_DEGREE>) {
     let remainder = mod_poly_ark_ref(dividend);
     *dividend = remainder;
 }
+
+/// Constructs and returns a new polynomial modulus used for the polynomial field, `X^[MAX_POLY_DEGREE] + 1`.
+/// This means that `X^[MAX_POLY_DEGREE] = -1`.
+///
+/// This is the canonical but un-reduced form of the modulus, because the reduced form is the zero polynomial.
+///
+/// TODO: work out how to generically make this a lazy static.
+/// Crates like `interned` or `lazy_static` might help, but we'll have to expand their macros and make them generic.
+#[cfg(any(test, feature = "benchmark"))]
+pub fn slow_new_poly_modulus<const MAX_POLY_DEGREE: usize>() -> Poly<MAX_POLY_DEGREE> {
```

Comment:
Unfortunately there's no easy way to cache this modulus for every degree that's actually used in the program. Since this is test-only code, we can fix this later.